docs/docs/legal/data-processing-agreement.mdx

```diff
@@ -1,113 +1,276 @@
-If possible, the Processor will promptly implement measures proposed in the notification.
-Insofar required the Processor will assist the Customer in notifying any applicable regulatory authority.
+Upon becoming aware of a Security Incident, ZITADEL shall inform Customer without undue delay and provide all such timely information and cooperation as Customer may require for the Customer to fulfil its data breach or cybersecurity incident reporting obligations under (and in accordance with the timescales required by) Applicable Data Protection Law.
+Customer shall further take all such measures and actions as are reasonable and necessary to investigate, contain, and remediate or mitigate the effects of the Security Incident, to the extent that the remediation is within ZITADEL's control, and shall keep Customer informed of all material developments in connection with the Security Incident.
+
+Notwithstanding anything to the contrary, ZITADEL's notification of or response to a Security Incident under this section will not be construed as an acknowledgment by ZITADEL of any fault or liability with respect to such Security Incident.
 
 ### Deletion or destruction after termination
 
-Upon Customer's request, the Processor shall delete personal data received after the end of the agreement, unless there is a legal obligation for the Processor to store or further process such data.
+Upon termination or expiry of the Agreement, ZITADEL shall (at the Customer’s election) destroy or return to the Customer all Customer Data (including all copies of the Customer Data) in its possession or control (including any Customer Data subcontracted to a third party for processing).
+This requirement shall not apply to the extent that ZITADEL is required by any applicable law to retain some or all Customer Data, in which case ZITADEL shall isolate and protect the Customer Data from any further processing except to the extent required by such law until deletion is possible.
+
+### Customer's information and audit rights
+
+To the extent required under Applicable Data Protection Law and on written request from the Customer, ZITADEL shall provide written responses (which may include audit report summaries/extracts) to all reasonable requests for information made by the Customer related to its processing of Customer Personal Data as necessary to confirm ZITADEL's compliance with this DPA.
+The Customer shall not exercise this right more than once in any twelve (12)-month rolling period, except (i) if and when required by instruction of a competent data protection or other regulatory authority; or (ii) if ZITADEL has experienced a Security Incident.
```

Comment:
> ... (ii) if ZITADEL has experienced a Security Incident.

Is there a regulation that demands this or is this "us being nice"?

Reply:
This came as input from the legal council. This does not seem to be part of the regulations, and it's anyways covered under the audit rights. We could limit the scope further in saying "if Zitadel has a Security Incident where Customer is directly impacted".